Source/Core/Core/IOS/Network/KD/NetKDRequest.cpp

```diff
@@ -239,8 +240,15 @@ void NetKDRequestDevice::SchedulerWorker(const SchedulerEvent event)
 {
   if (event == SchedulerEvent::Download)
   {
-    // TODO: Implement downloader part of scheduler
-    return;
+    u16 entry_index{};
+    std::optional<u8> subtask_id{};
+    NWC24::ErrorCode code = DetermineDownloadTask(&entry_index, &subtask_id);
+    if (code != NWC24::WC24_OK)
+      LogError(ErrorType::KD_Download, code);
```

Comment:
Should you keep going after this or just return early?

Reply:
I should have returned, yes.